src/smc-webapp/printing.coffee

```diff
@@ -525,6 +532,9 @@ class SagewsPrinter extends Printer
                 process_collected_mesg_stdout()
                 line++
             input_lines_process(final = true)
+            # combine all html snippets to one html block
+            @_html = (h for h in @_html).join('\n')
```

Comment:
yes, that's one of the things I had to change in order to find a way to do this with the video tag. the `process_lines` is only called once and I think it's best to drop `@_html` entirely.